teps/0142-enable-step-reusability.md

```diff
@@ -843,6 +844,14 @@ spec:
 </tr>
 </table>
 
+##### StepResults in Inlined-Steps
+
+To enable consistency of `results` usage in `StepActions`, we introduce a `Results` field to the `Step` struct which will be identical to the `Results` field in `StepActions`. `Task` authors cannot use this field at the same time as referencing a `StepAction` since in that case, the `results` from `StepActions` will fill it in. Using it simultaneously with `ref` will result in a validation error.
+
+However, if the `Task` author decides to produce `Step Results` for their inlined steps, they need do declare `Results` in `Steps`. Since this capability for inlined `Steps` is being introduced for consistency with `StepActions`, we gate it behind `enable-step-actions`.
+
+For fully, inlined `Tasks` (i.e. where all the `Steps` are inlined), the use of `Step Results` is not too useful. There have been occasional requests from users to provide the ability to pass `results` as `params` between `Steps`. Until now, we have asked the users to load the results from the files themselves in the `Steps` as a workaround. However, with `StepActions`, the need to pass `results` between `Steps` is necessary since there is no workaround for it. We already plan to do this to support `StepActions` so declaring the `Step Results` and making use of this ability even for `inlined Steps` is something `Task` authors can leverage. 
```

Comment:
maybe add a simple example to better show how results are passed between steps?

Reply:
Agreed. I can add an example to show it better.

Reply:
To show how results are passed between steps, actually it is shown in the `Results` section.

Reply:
This is just an added benefit that comes out of consistency, not a core requirement so I'm not sure if I should add an example showing this for inlined steps.

Reply:
WDYT?

Reply:
I did add an example showing how a Task could use Step Results in an inlined step to pass it as a parameter to a StepAction.